src/components/Navigation/components/Item/Item.tsx

```diff
@@ -137,14 +128,38 @@ export class BaseItem extends React.Component<CombinedProps, State> {
       )
     );
 
+    let badgeMarkup: Props['badge'] = null;
+    if (isNew) {
+      badgeMarkup = (
+        <Badge status="new" size="small">
+          {intl.translate('Polaris.Badge.STATUS_LABELS.new')}
+        </Badge>
+      );
+    }
+
+    if (badge != null && !isNew) {
```

Comment:
Can you do this in an `else` to the previous block to remove the `isNew` check?

Reply:
Yeah the only downside is if `badge` is undefined we'll be setting `badgeMarkup = undefined` but the later check handles this, and it's easier to read than nested if statements.